mlir/tools/rocmlir-gen/rocmlir-gen.cpp

```diff
@@ -3746,6 +3748,31 @@ int main(int argc, char **argv) {
   llvm::cl::ParseCommandLineOptions(argc, argv,
                                     "MLIR Rock Dialect host generation\n");
 
+  if (!arch.getValue().empty()) {
+    SmallVector<std::string>f8e4m3TypeNames{"", "f8E4M3FNUZ", "f8E4M3FN"};
+    SmallVector<std::string>f8e5m2TypeNames{"", "f8E5M2FNUZ", "f8E5M2"};
+    int archIndex = llvm::to_underlying(F8TypesChoice::Arch);
+    if (archChip().substr(0, 5) == "gfx12") {
```

Comment:
I don't like this approach either.

Could you instead, when `arch` is the fp8 type interpretation, just look at the architecture and then go to an if statement for nanoo/ocp? This hacking on arrays thing doesn't feel right. Especially given the amount of `to_underlying()` involved, which just feels wrong

Reply:
Went with a map<> to avoid the to_underlying() and ?: in the init list to avoid map-hacking.  Trying to simplify the levels of the conditionals.